### PR TITLE
XWIKI-17588: Display some text when there's no tag

### PR DESCRIPTION
**Pull request by CatalinStratu**
**- Display some text when there's no tag for guest**
Guest_user
 **-Display some text when there's no tag for logged user**
Logged_user

### CODE REVIEW

**Comment by vmassol** [comment]
Thanks for the PR. However it doesn't obey what's been decided on the jira issue in the comments (which is to not display anything for guest users when there are no tags). Could you update your PR? Thx

File: xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/documentTags.vm [hunk 1]
```diff
@@ -131,9 +131,13 @@
         #else
             ## TODO
         #end
+    #else
+       #if($tags.size() == 0)
```
Comment (vmassol) [inline xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/documentTags.vm:135]: why not `$tags.isEmpty()` which is the traditional way of doing it?

**Comment by vmassol** [comment]
Note that ideally we also need a test to prove that it works. You could check the existing tag tests to see if you could easily add one?

**Comment by CatalinStratu** [comment]
@vmassol 
**TEST**
**Logged user**
test_logged
**Guest User**
test_guest

File: xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/documentTags.vm (hunk 1 above)
Comment (CatalinStratu) [reply xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/documentTags.vm]: In new commit I use this

**Comment by vmassol** [comment]
> Thanks for the PR. However it doesn't obey what's been decided on the jira issue in the comments (which is to not display anything for guest users when there are no tags). Could you update your PR? Thx

This is still not done. This PR cannot be applied ATM. 

Note: It's also missing tests. What's your plan and strategy for that? Thx

File: xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/documentTags.vm [hunk 2]
```diff
@@ -131,9 +131,13 @@
         #else
             ## TODO
         #end
+    #else
+       #if($tags.isEmpty())
+        <div class="tag-tool tag-add"><a>None</a></div>
```
Comment (vmassol) [inline xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/documentTags.vm:136]: BTW why one space for indentation instead of 2?

**Comment by CatalinStratu** [comment]
Guest 0 tags
guest_no_tags

Guest one tag for test
guest_test_tag

Logged 1 tag
logged_no_tags

Logged 1 tag 
logged_1_tag

**Comment by CatalinStratu** [comment]
Hi, @vmassol  can you do a review on my PR?

**Comment by vmassol** [comment]
> Hi, @vmassol can you do a review on my PR?

Note: As I said it's missing some tests to prove it works. Normally any bug fix should have a test to prove that the issue is fixed and won't happen again. Since this is a vm file change, it's not that easy to write. So I'm trying to write a PageTest. Will take some time.

**Comment by vmassol** [comment]
I've now added a test, please verify if it still passes (it shouldn't!) and update it accordingly with your changes :)

https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-web/src/test/java/org/xwiki/web/DocumentTagsTest.java

**Comment by CatalinStratu** [comment]
> I've now added a test, please verify if it still passes (it shouldn't!) and update it accordingly with your changes :)
> 
> https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-web/src/test/java/org/xwiki/web/DocumentTagsTest.java

Thanks :))

**Comment by Enygma2002** [comment]
> Guest 0 tags
> guest_no_tags

This does not look right. Careful not to introduce new bugs ("regressions") in your attempt to improve/fix another.

The `Created by ...` text needs to stay in its place, in all cases.

**Comment by vmassol** [comment]
> This does not look right. Careful not to introduce new bugs ("regressions") in your attempt to improve/fix another.

I was suspecting this which is why I wrote a test for the documentTags.vm :) (which should fail with the PR).

**Pull request closed without merging**